From 48d1cc8e80d1a83baf456b75a5d624689075bd32 Mon Sep 17 00:00:00 2001
From: Valentin Caron <valentin.caron@foss.st.com>
Date: Wed, 12 Oct 2022 19:06:20 +0200
Subject: [PATCH 0070/1141] arm64: dts: st: add UART console on stm32mp257f-ev
 boards

Add pins and node for USART2 to support console on stm32mp257f-ev boards.

Signed-off-by: Valentin Caron <valentin.caron@foss.st.com>
Change-Id: I9c67450663bfd22f0e1f6817267ad318d5d2279a
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/270684
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Tested-by: Eric FOURMONT <eric.fourmont-ext@st.com>
Reviewed-by: Eric FOURMONT <eric.fourmont-ext@st.com>
Domain-Review: Eric FOURMONT <eric.fourmont-ext@st.com>
---
 arch/arm64/boot/dts/st/stm32mp25-pinctrl.dtsi | 31 +++++++++++++++++++
 arch/arm64/boot/dts/st/stm32mp257f-ev.dts     | 11 +++++++
 2 files changed, 42 insertions(+)

--- a/arch/arm64/boot/dts/st/stm32mp25-pinctrl.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp25-pinctrl.dtsi
@@ -5,3 +5,34 @@
  */
 #include <dt-bindings/pinctrl/stm32-pinfunc.h>
 
+&pinctrl {
+	usart2_pins_a: usart2-0 {
+		pins1 {
+			pinmux = <STM32_PINMUX('A', 4, AF6)>; /* USART2_TX */
+			bias-disable;
+			drive-push-pull;
+			slew-rate = <0>;
+		};
+		pins2 {
+			pinmux = <STM32_PINMUX('A', 8, AF8)>; /* USART2_RX */
+			bias-disable;
+		};
+	};
+
+	usart2_idle_pins_a: usart2-idle-0 {
+		pins1 {
+			pinmux = <STM32_PINMUX('A', 4, ANALOG)>; /* USART2_TX */
+		};
+		pins2 {
+			pinmux = <STM32_PINMUX('A', 8, AF8)>; /* USART2_RX */
+			bias-disable;
+		};
+	};
+
+	usart2_sleep_pins_a: usart2-sleep-0 {
+		pins {
+			pinmux = <STM32_PINMUX('A', 4, ANALOG)>, /* USART2_TX */
+				 <STM32_PINMUX('A', 8, ANALOG)>; /* USART2_RX */
+		};
+	};
+};
--- a/arch/arm64/boot/dts/st/stm32mp257f-ev.dts
+++ b/arch/arm64/boot/dts/st/stm32mp257f-ev.dts
@@ -17,6 +17,7 @@
 	compatible = "st,stm32mp257f-ev", "st,stm32mp257";
 
 	aliases {
+		serial0 = &usart2;
 	};
 
 	chosen {
@@ -33,3 +34,13 @@
 	timeout-sec = <32>;
 	status = "okay";
 };
+
+&usart2 {
+	pinctrl-names = "default", "idle", "sleep";
+	pinctrl-0 = <&usart2_pins_a>;
+	pinctrl-1 = <&usart2_idle_pins_a>;
+	pinctrl-2 = <&usart2_sleep_pins_a>;
+	/delete-property/dmas;
+	/delete-property/dma-names;
+	status = "okay";
+};
